### ISSUE TITLE
AzureAppInsights - WCF | Missing Telemetry data in Azure App Insights

### ISSUE THREAD

**Issue by sra2019** [microsoft/ApplicationInsights-Home#450]
Hello Folks,

WCF service using Microsoft.ApplicationInsights nuget v2.4.0:
A WCF service using AppInsights is not recording no of requests (RequestTelemetry) correctly. There is no issue with Dependency Telemetry. The no of requests were matching up when the service was hosted on 2008R2; but after a recent upgrade to 2019 server; once the cut over was done, ding! All the recorded requests traffic in AppInsights dropped by more than 60-70%. The concern is where are the missing requests or how to diagnose them since there are no application errors from the service end nor the customers have raised concerns on non responding service.

With a load test of 100K requests, they were captured / reflected correctly in test environment. But with a load of 200K requests, I noticed that the AppInsights has recorded only 199838 requests - where did the other 162 odd requests go is the puzzling piece here.

Any one experienced this issue using WCF - App Insights? Any thoughts on how to track this in Azure App Insights?

Thanks,
Sam.

**Comment by Dmitry-Matveev** [comment]
There might be several issues in play:
- AI .NET SDK 2.4 still has .NET 4.0 target, so it won't support TLS 1.2 out of the box and updating servers from 2008 to the latest 2019 may bring TLS 1.2 enforcement up;
- Application Insights SDKs come with sampling by default, if SDK sends more than 5 telemetry items per second it will re-evaluate sampling percentage to only send 5 telemetry items per second and will stamp each item the gain-up number on how to restore the actual count in UI. In such situation, pre-aggregated metrics that are calculated before sampling should be the most correct, the actual count of the events won't be 100% precise anymore is sampling kicks-in;
- WCF in itself is not fully supported by AI SDK by default, there is unofficial nuget on SDK labs with the better support, but that still has issues as well. For instance, you can see [this set of issues](https://github.com/microsoft/ApplicationInsights-dotnet/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+wcf) with WCF that are still opened.

**Comment by sra2019** [comment]
Thanks Dmitry - your inputs helped. Really appreciate your kind gesture to guide. Am surprised Microsoft abandoned wcf supportability for appInsights. The referred gitHub project never got updated from last 3 years. Looks like WCF is not favored service stack of Microsoft anymore. 

That said - we identified the issue ; the memory leak in another thread of the app was spiking up memory of the app pool - there by pushing the implemented telemetry client to wait for return from that thread to push data to appInsights. 

Thanks,
Shamant
